### PR TITLE
Add basic UI test with Spectron

### PR DESCRIPTION
**Pull request by NazulIslam**
Basic UI test with Spectron.

### CODE REVIEW

File: examples/electron/test/basic-example-with-spectron.espec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    startTimeout: timeout,
+    waitTimeout: timeout,
+    requireName: 'electronRequire'
+});
+
+describe('Test specs with spectron', function () {
+    this.timeout(60000);
+
+    beforeEach(function () {
+        require("../src-gen/backend/main"); // start the express server
```
Comment (akosyakov) [inline examples/electron/test/basic-example-with-spectron.espec.ts]: Should not it be done by `electron-main` and we only should wait until the window is visible?

Comment (NazulIslam) [inline examples/electron/test/basic-example-with-spectron.espec.ts]: I tried to add waitUntilWindowLoaded(), but got an error: 
        return app.client.waitUntilWindowLoaded(10000)

     TypeError: waitUntilWindowLoaded Cannot read property 'isLoading' of undefined
      at waitUntil(<Function>, 60000) - application.js:227:17

error Command failed with exit code 1.

Looking at https://github.com/electron/spectron/issues/174, seems similar.

**Comment by hexa00** [comment]
See #87 for the devtools issue

**Comment by svenefftinge** [comment]
can this be closed, because you mentioned yesterday you stop looking into it.

**Pull request closed without merging**